pkg/azuredx/adxauth/adxusercontext/from_req.go

```diff
@@ -0,0 +1,98 @@
+package adxusercontext
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+func FromQueryReq(ctx context.Context, req *backend.QueryDataRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	idToken := getQueryReqHeader(req, "X-ID-Token")
+	accessToken := extractBearerToken(getQueryReqHeader(req, "Authorization"))
+
+	currentUser := CurrentUserContext{
+		User:        req.PluginContext.User,
+		IdToken:     idToken,
+		AccessToken: accessToken,
+	}
+
+	return WithCurrentUser(ctx, currentUser)
+}
+
+func FromResourceReq(ctx context.Context, req *backend.CallResourceRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	idToken := getResourceReqHeader(req, "X-ID-Token")
+	accessToken := extractBearerToken(getResourceReqHeader(req, "Authorization"))
+
+	currentUser := CurrentUserContext{
+		User:        req.PluginContext.User,
+		IdToken:     idToken,
+		AccessToken: accessToken,
+	}
+
+	return WithCurrentUser(ctx, currentUser)
+}
+
+func FromHealthCheckReq(ctx context.Context, req *backend.CheckHealthRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	// TODO: Add support for ID and access tokens
+	currentUser := CurrentUserContext{
+		User:        req.PluginContext.User,
+		IdToken:     "",
+		AccessToken: "",
+	}
+
+	return WithCurrentUser(ctx, currentUser)
+}
+
+func getQueryReqHeader(req *backend.QueryDataRequest, headerName string) string {
+	// TODO: Check what subset of Unicode is case-insensitive for request headers, possibly narrow to ASCII
```

Comment:
I would either remove this TODO or resolve it (doesn't sound like it's blocked by anything).

Reply:
Yes, since these funcs are private and used only for specific headers `X-ID-Token`, `Authorization`, there's no need to worry about case sensitivity and Unicode.